docs/src/intro/usage.md

```diff
@@ -51,4 +52,137 @@ In the `miden/examples/fib` directory, we provide a very simple Fibonacci calcul
 ```
 ./target/release/miden run -a miden/examples/fib/fib.masm -n 1
 ```
-This will run the example code to completion and will output the top element remaining on the stack.
\ No newline at end of file
+This will run the example code to completion and will output the top element remaining on the stack.
+
+## REPL
+
+The Miden Read–eval–print loop (REPL) is a Miden shell that allows for quick and easy debugging of Miden assembly. After the REPL gets initialized, you can execute any Miden instruction, undo executed instructions, check the state of the stack and memory at a given point, and do many other useful things! When the REPL is exited, a `history.txt` file is saved. One thing to note is that all the REPL native commands start with an `!` to differentiate them from regular assembly instructions. The REPL currently supports the following commands:
+
+### Miden assembly instruction
+
+All Miden instructions mentioned in the [Miden Assembly sections](../user_docs/assembly/main.md) are valid. One can either input instructions one by one or multiple instructions in one input.
+
+For example, the below two commands will result in the same output.
+```
+>> push.1
+>> push.2
+>> push.3
+```
+
+```
+push.1 push.2 push.3
+```
+
+To execute a control flow operation, one must write the entire statement in a single line with spaces between individual operations.
+
+```
+repeat.20
+    pow2
+end
+```
+
+The above example should be written as follows in the REPL tool:
+
+```
+repeat.20 pow2 end
+```
+
+### !help
+
+The `!help` command prints out all the available commands in the REPL tool. 
+
+### !program
+
+The `!program` command prints out the entire miden program getting executed. E.g., in the below scenario:
```

Comment:
nit: "Miden" should be capitalized.

I would also say "being executed" rather than "getting executed."